phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -903,7 +903,7 @@ private boolean addColumnsAndIndexesFromAncestors(MetaDataMutationResult result,
                 String parentSchemaName = SchemaUtil.getSchemaNameFromFullName(parentName);
                 tableName = SchemaUtil.getTableNameFromFullName(parentName);
                 MetaDataMutationResult parentResult = updateCache(connection.getTenantId(), parentSchemaName, tableName,
-                        false, resolvedTimestamp);
+                        true, resolvedTimestamp);
```

Comment:
Why is this change needed?

Reply:
@shahrs87  `MetaDataClient.updateCache` eventually calls this method `addColumnsAndIndexesFromAncestors` which looks up ancestors of views and view indexes and recursively calls `MetaDataClient.updateCache` on every parent table up the hierarchy. Currently, the value of `alwaysHitServer` is false for all these recursive calls i.e. we will look up the ancestors in the local cache if UPDATE_CACHE_FREQUENCY is set to NEVER. Setting it to true bypasses all checks for avoiding the rpc to get table ([code](https://github.com/apache/phoenix/blob/master/phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java#L749)) and updates the entire hierarchy.

Reply:
I think to avoid any side effects, it will be better to pass the `alwaysHitServer` value to `addColumnsAndIndexesFromAncestors` instead of just setting it to true always. I will make that change.